plugins/admin/pkg/command/profiling/profiling.go

```diff
@@ -0,0 +1,458 @@
+				flags.Changed(goroutineFlagName) || flags.Changed(threadCreateFlagName))
+
+			// enable and disable can't be used togerther
+			if isEnableSet && isDisableSet {
+				return fmt.Errorf("flags '--enable' and '--disable' can not be used together")
+			}
+
+			// enable or disable can't be used with other flags
+			if (isEnableSet || isDisableSet) && (isTargetSet || isProfileTypeSet || isAllProfilesSet || isSaveToSet) {
+				return fmt.Errorf("flag '--enable' or '--disable' can not be used with other flags")
+			}
+
+			// --target flag is needed
+			if !isTargetSet && (isProfileTypeSet || isAllProfilesSet || isSaveToSet) {
+				return fmt.Errorf("requires '--target' flag")
+			}
+
+			// --profile-type is needed
+			if !isProfileTypeSet && !isAllProfilesSet && (isTargetSet || isSaveToSet) {
+				return fmt.Errorf("requires '--all' or a specific profile type flag")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			flags := cmd.Flags()
+			if flags.NFlag() < 1 {
+				return nil
+			} else if flags.Changed("enable") {
+				return configProfiling(p.ClientSet, cmd, true)
+			} else if flags.Changed("disable") {
+				return configProfiling(p.ClientSet, cmd, false)
+			} else {
+				return downloadProfileData(p, cmd, &pflags)
+			}
+		},
+	}
+
+	flags := profilingCmd.Flags()
+	flags.BoolVar(&pflags.enable, "enable", false, "Enable Knative profiling")
+	flags.BoolVar(&pflags.disable, "disable", false, "Disable Knative profiling")
+	flags.StringVarP(&pflags.target, "target", "t", "", targetFlagUsgae)
+	flags.StringVarP(&pflags.saveTo, "save-to", "s", "", saveToFlagUsage)
+	flags.BoolVar(&pflags.allProfiles, "all", false, allFlagUsage)
+	flags.StringVarP(&pflags.cpuProfile, cpuFlagName, "", "5s", cpuFlagUsage)
+	flags.BoolVar(&pflags.heapProfile, heapFlagName, false, heapFlagUsage)
+	flags.BoolVar(&pflags.blockProfile, blockFlagName, false, blockFlagUsage)
+	flags.StringVarP(&pflags.traceProfile, traceFlagName, "", "5s", traceFlagUsage)
+	flags.BoolVar(&pflags.memAllocsProfile, memAllocsFlagName, false, memAllocsFlagUsage)
+	flags.BoolVar(&pflags.mutexProfile, mutexFlagName, false, mutexFlagUsage)
+	flags.BoolVar(&pflags.goroutineProfile, goroutineFlagName, false, goroutineFlagUsage)
+	flags.BoolVar(&pflags.threadCreateProfile, threadCreateFlagName, false, threadCreateFlagUsage)
+	return profilingCmd
+}
+
+// configProfiling enables or disables knative profiling
+func configProfiling(c kubernetes.Interface, cmd *cobra.Command, enable bool) error {
+	currentCm := &corev1.ConfigMap{}
+	currentCm, err := c.CoreV1().ConfigMaps(knNamespace).Get(obsConfigMap, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get ConfigMap %s in namespace %s: %+v", obsConfigMap, knNamespace, err)
+	}
+
+	desiredCm := currentCm.DeepCopy()
+	if enable {
+		desiredCm.Data["profiling.enable"] = "true"
+	} else {
+		desiredCm.Data["profiling.enable"] = "false"
+	}
+
+	err = utils.UpdateConfigMap(c, desiredCm)
```

Comment:
there's a common utils can help update ConfigMap, please help to update this part to the `func UpdateConfigMap(client kubernetes.Interface, desiredCm *corev1.ConfigMap)`

Reply:
I think it is using the utils package already. Did I miss something?

Reply:
Last time I remember it is using client.coreV1()
Anyway, it looks good now.